pkg/webui/console/views/device-add/index.js

```diff
@@ -0,0 +1,324 @@
+        <Field
+          title={sharedMessages.fwdNtwkKey}
+          name="session.keys.f_nwk_s_int_key.key"
+          type="byte"
+          min={16}
+          max={16}
+          placeholder={m.leaveBlankPlaceholder}
+          description={m.fwdNtwkKeyDescription}
+        />
+        <Field
+          title={sharedMessages.sNtwkSIKey}
+          name="session.keys.s_nwk_s_int_key.key"
+          type="byte"
+          min={16}
+          max={16}
+          placeholder={m.leaveBlankPlaceholder}
+          description={m.sNtwkSIKeyDescription}
+        />
+        <Field
+          title={sharedMessages.ntwkSEncKey}
+          name="session.keys.nwk_s_enc_key.key"
+          type="byte"
+          min={16}
+          max={16}
+          placeholder={m.leaveBlankPlaceholder}
+          description={m.ntwkSEncKeyDescription}
+        />
+        <Field
+          title={sharedMessages.appSKey}
+          name="session.keys.app_s_key.key"
+          type="byte"
+          min={16}
+          max={16}
+          placeholder={m.leaveBlankPlaceholder}
+          description={m.appSKeyDescription}
+        />
+      </React.Fragment>
+    )
+  }
+
+  OTAASection () {
+    return (
+      <React.Fragment>
+        <Field
+          title={sharedMessages.joinEUI}
+          name="ids.join_eui"
+          type="byte"
+          min={8}
+          max={8}
+          placeholder={m.joinEUIPlaceholder}
+          required
+        />
+        <Field
+          title={sharedMessages.devEUI}
+          name="ids.dev_eui"
+          type="byte"
+          min={8}
+          max={8}
+          placeholder={m.deviceEUIPlaceholder}
+          description={m.deviceEUIDescription}
+          required
+        />
+        <Field
+          title={sharedMessages.nwkKey}
+          name="root_keys.nwk_key.key"
+          type="byte"
+          min={16}
+          max={16}
+          placeholder={m.leaveBlankPlaceholder}
+          description={m.nwkKeyDescription}
+
+        />
+        <Field
+          title={sharedMessages.appKey}
+          name="root_keys.app_key.key"
+          type="byte"
+          min={16}
+          max={16}
+          placeholder={m.leaveBlankPlaceholder}
+          description={m.appKeyDescription}
+        />
+        <Field
+          title={m.resetsJoinNonces}
+          name="resets_join_nonces"
+          type="checkbox"
+        />
+      </React.Fragment>
+    )
+  }
+
+  render () {
+    const { error, otaa } = this.state
+
+    return (
+      <Container>
+        <Row className={style.wrapper}>
+          <Col sm={12}>
+            <IntlHelmet title={sharedMessages.addDevice} />
+            <Message component="h2" content={sharedMessages.addDevice} />
+          </Col>
+          <Col className={style.form} sm={12} md={12} lg={8} xl={8}>
+            <Form
+              error={error}
+              onSubmit={this.handleSubmit}
+              validationSchema={validationSchema}
+              submitEnabledWhenInvalid
```

Comment:
Any reasons to enable submission for an invalid form?

Reply:
If this is not enabled there are two problems:
1. If we set `validateOnBlur` to true, the button will stay disabled even when all fields are properly filled, because the user needs to blur out of the last one, resulting in bad UX
2. If we set `validateOnChange` to true, the issue above is resolved but the form becomes very slow because it has to run the complex validation logic on every keystroke

I did not want to spend too much time on this, as having the submit button enabled and showing respective field errors when it is pressed is sufficiently good, UX wise. We should look into ways to fix this later though.

Reply:
I would expect a new issue related to this

Reply:
⟶ #582